source/extensions/filters/network/common/redis/fault_impl.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.pb.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/network/common/redis/fault.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class Fault {
+public:
+  Fault(envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault);
+
+private:
+  static std::vector<std::string> getCommands(
+      envoy::extensions::filters::network::redis_proxy::v3::RedisProxy_RedisFault base_fault);
+
+public:
+  FaultType fault_type_;
+  std::chrono::milliseconds delay_ms_;
+  const std::vector<std::string> commands_;
+  bool has_fault_enabled_;
+  absl::optional<int> default_value_;
```

Comment:
what do we do with unset default_value_ and runtime_key_?

Reply:
I'm changing it for default_value_ to not be optional and be zero by default.